aries_cloudagent/protocols/present_proof/dif/pres_exch_handler.py

```diff
@@ -1187,12 +1206,18 @@ async def create_vp(
     def check_sign_pres(self, creds: Sequence[VCRecord]) -> bool:
         """Check if applicable creds have CredentialSubject.id set."""
         for cred in creds:
-            if len(cred.subject_ids) > 0 and not next(
-                iter(cred.subject_ids)
-            ).startswith("urn:"):
+            if len(cred.subject_ids) > 0 and not self.check_if_cred_id_derived(
+                next(iter(cred.subject_ids))
+            ):
                 return True
         return False
 
+    def check_if_cred_id_derived(self, id: str) -> bool:
+        """Check if credential or credentialSubjet id is derived."""
+        if id.startswith("urn:bnid:_:c14n"):
```

Comment:
This seems too specific to me, wouldn't a different blank node identifier be generated depending on the ordering of the normalized statements?

Reply:
When deriving, I think all blank node identifiers are converted into `urn:bnid:_:c14nX` node ids, @TimoGlastra any suggestions?

Reply:
So the [BBS+ Signatures 2020 draft](https://w3c-ccg.github.io/ldp-bbs2020/) doesn't seem to be explicit about the format of the blank node identifiers. I think it's trying to say that it's covered by the canonicalization algorithm, but the link is broken. Assuming this is the the correct link, it doesn't really specify the format of the identifiers either although it gives some guidelines:
https://json-ld.github.io/rdf-dataset-canonicalization/spec/#blank-node-identifier-issuer-state

In any case, I think the prefix you're using is probably consistent with the current implementations.

Reply:
This is indeed how current implementations handle it: https://github.com/w3c-ccg/ldp-bbs2020/issues/10#issuecomment-616216278

I'm working on a PR to add this to the spec